Viper platform fixes

raveenp opened the pull request:
This PR contains following Viper platform fixes:
- soc: viper: update registers and interrupt numbers
- soc: arm: bcm_vk: viper: Increase default NUM_IRQS

raveenp committed:
soc: viper: update registers and interrupt numbers

Update register addresses and NVIC/GIC interrupt numbers
according to the latest viper RTL version.

Signed-off-by: Raveendra Padasalagi <raveendra.padasalagi@broadcom.com>
Signed-off-by: Abhishek Shah <abhishek.shah@broadcom.com>

galak reviewed (commented):
It would probably be best to split this into at least 3 PRs:
* dma changes / fixes
* iproc changes / fixes
* platform changes / fixes

galak commented on boards/arm/bcm958402m2_a72/bcm958402m2_a72_defconfig:
```diff
@@ -25,3 +25,9 @@ CONFIG_UART_CONSOLE=y
+
+# Enable PAXDMA
+CONFIG_DMA=y
+CONFIG_DMA_64BIT=y
```
What about DMA_PL330?

raveenp replied:
Will enable in https://github.com/zephyrproject-rtos/zephyr/pull/32316

galak commented on boards/arm/bcm958402m2_m7/bcm958402m2_m7_defconfig:
```diff
@@ -18,3 +18,13 @@ CONFIG_UART_NS16550=y
+
+# Enable PAXDMA
+CONFIG_DMA=y
+CONFIG_DMA_IPROC_PAX_V2=y
```
What about DMA_PL330?

raveenp replied:
Will enable in https://github.com/zephyrproject-rtos/zephyr/pull/32316

olofj commented:
Besides the comments from Kumar, please also provide better descriptions in the commits (or in the Pull request). The "why" is missing -- in particular why basic changes such as remapping of interrupts is needed, or why they were wrong before.

raveenp commented:
> It would probably be best to split this into at least 3 PRs:
> 
> * dma changes / fixes
> * iproc changes / fixes
> * platform changes / fixes

Sure, I will split the patch and push in separate PR's.

raveenp commented:
> Besides the comments from Kumar, please also provide better descriptions in the commits (or in the Pull request). The "why" is missing -- in particular why basic changes such as remapping of interrupts is needed, or why they were wrong before.

Sure, will provide the descriptions in the commit.

raveenp committed:
soc: arm: bcm_vk: viper: Increase default NUM_IRQS

Maximum GIC SPI is 228, so maximum INTID is 228 + 32 = 260.
So, _sw_isr_table needs to support 260 entries

Signed-off-by: Raveendra Padasalagi <raveendra.padasalagi@broadcom.com>
Signed-off-by: Abhishek Shah <abhishek.shah@broadcom.com>
